src/main/java/com/codeplanks/home360/auth/AuthenticationService.java

```diff
@@ -0,0 +1,69 @@
+package com.codeplanks.home360.auth;
+
+
+import com.codeplanks.home360.config.JwtService;
+import com.codeplanks.home360.exception.UserAlreadyExistsException;
+import com.codeplanks.home360.user.AppUser;
+import com.codeplanks.home360.user.IUserService;
+import com.codeplanks.home360.user.Role;
+import com.codeplanks.home360.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService implements IUserService {
+    private final UserRepository userRepository;
+    private final JwtService jwtService;
+    private final PasswordEncoder passwordEncoder;
+    private final AuthenticationManager authenticationManager;
+
+    public AuthenticationResponse register(@RequestBody RegisterRequest request) throws UserAlreadyExistsException {
```

Comment:
There's no need to add the RequestBody annotation in the service, it is only needed in the controller.